focstest.py

```diff
@@ -215,7 +216,7 @@ def format_info(kind, value):
     # TODO: get titles/descriptions from code blocks
     blocks = get_blocks(html)
     # parse code blocks for tests
-    test_suites = list(enumerate([get_tests(block) for block in blocks], 1))  # list of suites and indices (starting at 1)
+    test_suites = list(enumerate([get_tests(block) for block in blocks if get_tests(block)], 1))  # list of suites and indices (starting at 1) (skipping empty suites)
```

Comment:
Instead of running `get_tests` twice, what about filtering for empty arrays?
I have also learned the difference between generators and list comprehensions since writing this code, and think this would benefit from the former.

Something like:
```python
list(enumerate(filter(None, (get_tests(block) for block in blocks)), 1))
```

Reply:
I agree with these and added them